user_guide_src/source/incoming/controllers.rst

```diff
@@ -112,6 +112,7 @@ rule and message array formats, as well as available rules:
     or :ref:`$request->getRawInput() <incomingrequest-retrieving-raw-data>`
     or :ref:`$request->getVar() <incomingrequest-getting-data>`, and an attacker
     could change what data is validated.
+    Instead of withRequest(), use validateData() to validate POST data only.       withRequest() uses $request->getVar() which returns $_GET, $_POST and $_COOKIE data in that order. Newer values override older values. Post values will be overriden by the cookies if they have the same name.
```

Comment:
This warning will be too long. 
Why don't you add another warning about this addition, and move it to the top of the section?
Personally `$this->validate()` is not recommended.

```suggestion
    Instead of ``validate()``, use ``validateData()`` to validate POST data only.
    ``withRequest()`` uses ``$request->getVar()`` which returns ``$_GET``, ``$_POST``
    or ``$_COOKIE`` data in that order (depending on php.ini
    `request-order <https://www.php.net/manual/en/ini.core.php#ini.request-order>`_).
    Newer values override older values. POST values may be overridden by the
    cookies if they have the same name.
```

Reply:
Thanks for the edit. That is what I was trying to explain. But unfortunately I don't know this markup language used here. I would appreciate if you could do it, please.

Reply:
See here it might help you.
https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/documentation.rst

Reply:
Okay, I sent #8169